packages/block/src/constructors.ts

```diff
@@ -515,3 +526,31 @@ export async function createBlockFromBeaconPayloadJson(
   const executionPayload = executionPayloadFromBeaconPayload(payload)
   return createBlockFromExecutionPayload(executionPayload, opts)
 }
+
+/**
+ *  Method to retrieve a sealed block from the header and clique signer key
+ * @param header block header to be used for creating sealed block
+ * @param cliqueSigner clique signer key to be used for creating sealed block
+ * @returns the block constructed block
+ */
+export function createSealedCliqueBlock(header: BlockHeader, cliqueSigner: Uint8Array): Uint8Array {
+  // Ensure extraData is at least length CLIQUE_EXTRA_VANITY + CLIQUE_EXTRA_SEAL
+  const minExtraDataLength = CLIQUE_EXTRA_VANITY + CLIQUE_EXTRA_SEAL
+  if (header.extraData.length < minExtraDataLength) {
+    const remainingLength = minExtraDataLength - header.extraData.length
+    ;(header.extraData as any) = concatBytes(header.extraData, new Uint8Array(remainingLength))
+  }
+
+  _requireClique(header, 'cliqueSealBlock')
+
+  const ecSignFunction = header.common.customCrypto?.ecsign ?? ecsign
+  const signature = ecSignFunction(cliqueSigHash(header), cliqueSigner)
+  const signatureB = concatBytes(signature.r, signature.s, bigIntToBytes(signature.v - BIGINT_27))
+
+  const extraDataWithoutSeal = header.extraData.subarray(
+    0,
+    header.extraData.length - CLIQUE_EXTRA_SEAL,
+  )
+  const extraData = concatBytes(extraDataWithoutSeal, signatureB)
+  return extraData
+}
```

Comment:
Ah, no, this is not a constructor (and like this is also has no tree shaking effects, since this method is then used again in the main class).

So a constructor here would be a something with a signature like:

```ts
createSealedCliqueBlock(blockData: BlockData = {}, cliqueSigner: Uint8Array, opts?: BlockOptions): Block
```

And this method would calculate the signature (like above) and integrate this into `extraData` and then likely (?) overwrites the `extraData` from `blockData` and then creates and returns a block with this.

Respectively there should likely be a corresponding `createSealedCliqueBlockHeader()` method where all this stuff happens.

Just looking at the other constructors: we definitely do not want to copy-paste the `createBlock()` or so code here. So if we find a solution it should be one which has not significantly more code lines than the code above.

So I guess what we can do is:

1. Do this `extraData` calculation from above and...
2. then relatively directly hand this over to the generic `createBlock()` (`createHeader()`) method.

Guess that should work? 

Can you experiment with that?